plugin/plugin.go

```diff
@@ -0,0 +1,380 @@
+			return nil
+		}
+	} else {
+		return handleResult(<-done, args)
+	}
+}
+
+func resolveUploadFileList(includes, excludes string) (string, error) {
+	workspace := os.Getenv("DRONE_WORKSPACE")
+	if workspace == "" {
+		var err error
+		workspace, err = os.Getwd()
+		if err != nil {
+			return "", fmt.Errorf("could not determine workspace: %w", err)
+		}
+	}
+
+	matches, err := getAntMatchedFiles(workspace, includes, excludes)
+	if err != nil {
+		return "", fmt.Errorf("failed to match files: %w", err)
+	}
+
+	if len(matches) == 0 {
+		return "", fmt.Errorf("no files matched for upload")
+	}
+
+	return strings.Join(matches, ","), nil
+}
+
+func getAntMatchedFiles(root, includePatterns, excludePatterns string) ([]string, error) {
+	var results []string
+	includes := splitPatterns(includePatterns)
+	excludes := splitPatterns(excludePatterns)
+
+	matcher := antpathmatcher.NewAntPathMatcher()
+
+	err := filepath.WalkDir(root, func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+
+		// Get path relative to root for matching
+		relPath, err := filepath.Rel(root, path)
+		if err != nil {
+			return err
+		}
+		relPath = filepath.ToSlash(relPath) // normalize for matcher
+
+		// Check inclusion
+		included := false
+		for _, pat := range includes {
+			if matcher.Match(pat, relPath) {
+				included = true
+				break
+			}
+		}
+		if !included {
+			return nil
+		}
+
+		// Check exclusion
+		for _, pat := range excludes {
+			if matcher.Match(pat, relPath) {
+				return nil
+			}
+		}
+
+		results = append(results, filepath.Join(root, relPath))
+		return nil
+	})
+
+	return results, err
+}
+
+func splitPatterns(patterns string) []string {
+	var result []string
+	for _, p := range strings.Split(patterns, ",") {
+		if trimmed := strings.TrimSpace(p); trimmed != "" {
+			result = append(result, trimmed)
+		}
+	}
+	return result
+}
+
+func handleResult(err error, args Args) error {
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			output := string(exitError.Stderr)
+
+			// Check if version already exists
+			if strings.Contains(output, "already exists") {
+				logrus.Warnf("⚠️ The version '%s' already exists on Veracode. Please delete the existing version or use a new version string.", args.Version)
+			} else {
+				logrus.Errorf("❌ UploadAndScan failed with error: %s", output)
+			}
+		} else {
+			logrus.Errorf("❌ UploadAndScan failed: %v (job not marked as failed)", err)
+		}
+
+		if args.CanFailJob {
+			return fmt.Errorf("❌ UploadAndScan failed: %w", err)
+		}
+	} else {
+		logrus.Info("✅ UploadAndScan completed successfully")
+	}
+	return nil
+}
+
+func waitForScanCompletion(ctx context.Context, args Args) error {
+	logrus.Info("⏳ Waiting for scan to complete...")
+
+	timeout := time.After(time.Duration(args.Timeout) * time.Minute)
+	tick := time.Tick(1 * time.Minute)
```

Comment:
```suggestion
	ticker := time.NewTicker(1 * time.Minute)
        defer ticker.Stop()
```

then use ticker.C instead of tick in the select statement